File: benchmarks/realworld/compress/Go/compress.go
package main

import (
	"fmt"
	"os"
	"time"
)

func nowSec() float64 {
	return float64(time.Now().UnixNano()) / 1e9
}

func simpleCompress(input []byte) []byte {
	output := make([]byte, 0, len(input)*2)
	i := 0
	for i < len(input) {
		b := input[i]
		count := 1
		for i+count < len(input) && input[i+count] == b && count < 255 {
			count++
		}
		output = append(output, byte(count))
		output = append(output, b)
		i += count
	}
	return output
}

func main() {
	if len(os.Args) != 2 {
		fmt.Printf("Usage: %s <file>\n", os.Args[0])
		os.Exit(1)
	}

	inputData, err := os.ReadFile(os.Args[1])
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error reading file: %v\n", err)
		os.Exit(1)
	}

	tStart := nowSec()
	outputData := simpleCompress(inputData)
	tEnd := nowSec()

	fmt.Printf("Compression time: %.6f sec\n", tEnd-tStart)
	fmt.Printf("Original size: %d bytes\n", len(inputData))
	fmt.Printf("Compressed size: %d bytes\n", len(outputData))
}

File: benchmarks/realworld/compress/Python/compress.py
import time

def now_sec():
    return time.monotonic()

def simple_compress(data):
    out = bytearray()
    i = 0
    while i < len(data):
        b = data[i]
        count = 1
        while i + count < len(data) and data[i + count] == b and count < 255:
            count += 1
        out.append(count)
        out.append(b)
        i += count
    return out

if __name__ == "__main__":
    import sys
    if len(sys.argv) != 2:
        print(f"Usage: {sys.argv[0]} <file>")
        sys.exit(1)

    with open(sys.argv[1], "rb") as f:
        input_data = f.read()

    t_start = now_sec()
    output_data = simple_compress(input_data)
    t_end = now_sec()

    print(f"Compression time: {t_end - t_start:.6f} sec")
    print(f"Original size: {len(input_data)} bytes")
    print(f"Compressed size: {len(output_data)} bytes")
